src/components/activity-card/activity-card.tsx

```diff
@@ -42,14 +41,18 @@ const RecentEvent: React.FC<any> = (props) => {
   const [events, eventsLoaded] = useK8sWatchResource(eventsResource);
   const FlashsystemEventFilter = (event: EventKind): boolean => {
     const eventSource = event?.source?.component;
+
     const isIBMStorageCSIprovisioner =
       eventSource?.indexOf(IBM_STORAGE_CSI_PROVISIONER) != -1;
+
     const isFlashsystemClusterKind =
       eventSource?.indexOf(StorageInstanceModel.kind) != -1;
-    const eventName = _.get(event, ["metadata", "name"]);
-    const isNameIncluded = name ? eventName?.indexOf(name) != -1 : false;
+
+    const eventInvolveObjectName = event?.involvedObject?.name
+    const isNameIncluded = name ? eventInvolveObjectName == name : false;
+
     return (
-      isFlashsystemClusterKind || isIBMStorageCSIprovisioner || isNameIncluded
+      isFlashsystemClusterKind && isIBMStorageCSIprovisioner && isNameIncluded
```

Comment:
@bvered I am not sure which condition is better here. 
'OR' is not relevant I think, because we do not want to get events of all FlashSystemCluster objects, but I am not sure regarding 'AND', and also why do we need to filter by CSI events?
What do you think?

Reply:
fixed